cli/src/main/java/org/apache/iotdb/tool/ExportTsFile.java

```diff
@@ -0,0 +1,359 @@
+            .desc("Export file name (optional)")
+            .build();
+    options.addOption(targetFileName);
+
+    Option opSqlFile =
+        Option.builder(SQL_FILE_ARGS)
+            .argName(SQL_FILE_NAME)
+            .hasArg()
+            .desc("SQL File Path (optional)")
+            .build();
+    options.addOption(opSqlFile);
+
+    Option opQuery =
+        Option.builder(QUERY_COMMAND_ARGS)
+            .argName(QUERY_COMMAND_NAME)
+            .hasArg()
+            .desc("The query command that you want to execute. (optional)")
+            .build();
+    options.addOption(opQuery);
+
+    Option opHelp =
+        Option.builder(HELP_ARGS)
+            .longOpt(HELP_ARGS)
+            .hasArg(false)
+            .desc("Display help information")
+            .build();
+    options.addOption(opHelp);
+
+    return options;
+  }
+
+  /**
+   * This method will be called, if the query commands are written in a sql file.
+   *
+   * @param filePath:file path
+   * @throws IOException: exception
+   */
+  private static void dumpFromSqlFile(String filePath) throws IOException {
+    try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+      String sql;
+      int i = 0;
+      while ((sql = reader.readLine()) != null) {
+        legalCheck(sql);
+        dumpResult(sql, i++);
+      }
+    }
+  }
+
+  /**
+   * Dump files from database to tsFile.
+   *
+   * @param sql export the result of executing the sql
+   */
+  private static void dumpResult(String sql, int index) {
+    final String path = targetDirectory + targetFile + index + ".tsfile";
+    try {
+      SessionDataSet sessionDataSet = session.executeQueryStatement(sql, 10000);
+      writeTsFileFile(sessionDataSet, path);
+      sessionDataSet.closeOperationHandle();
+      System.out.println("Export completely!");
+    } catch (StatementExecutionException
+        | IoTDBConnectionException
+        | IOException
+        | WriteProcessException e) {
+      System.out.println("Cannot dump result because: " + e.getMessage());
+    }
+  }
+
+  public static void writeTsFileFile(SessionDataSet sessionDataSet, String filePath)
+      throws IOException, IoTDBConnectionException, StatementExecutionException,
+          WriteProcessException {
+    List<String> columnNames = sessionDataSet.getColumnNames();
+    List<String> columnTypes = sessionDataSet.getColumnTypes();
+    File f = FSFactoryProducer.getFSFactory().getFile(filePath);
+    if (f.exists()) {
+      f.delete();
+    }
+    try (TsFileWriter tsFileWriter = new TsFileWriter(f)) {
+      Map<String, Tablet> tabletMap = new LinkedHashMap<>();
+      for (int i = 0; i < columnNames.size(); i++) {
+        String column = columnNames.get(i);
+        int lastIndex = column.lastIndexOf(".");
+        if (lastIndex == -1) {
+          continue;
+        }
+        TSDataType tsDataType = getTsDataType(columnTypes.get(i));
+        String sensor = column.substring(lastIndex + 1);
```

Comment:
We can use this to split device and measurement.
https://github.com/apache/iotdb/blob/150686d0bd3c444ebffb3ad19e85d3f05f6911a6/tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/Path.java#L61-L66